File: backend-scala/src/main/scala/com/example/modules/payments/publicapi/PaymentsPublicApi.scala
// ONLY interface other modules/apps may import from the payments module.
package com.example.modules.payments.publicapi

import com.example.modules.payments.internal.{Payment, PaymentService}

class PaymentsPublicApi:
  private val svc = new PaymentService
  def authorize(paymentId: String, orderId: String, amountCents: Long): Payment = svc.authorize(paymentId, orderId, amountCents)
  def capture(paymentId: String): Option[Payment] = svc.capture(paymentId)
  def refund(paymentId: String): Option[Payment] = svc.refund(paymentId)

File: backend-kotlin/src/main/kotlin/com/example/modules/payments/publicapi/PaymentsPublicApi.kt
// ONLY interface other modules/apps may import from the payments module.
package com.example.modules.payments.publicapi

import com.example.modules.payments.internal.PaymentService

class PaymentsPublicApi {
    private val svc = PaymentService()
    fun authorize(paymentId: String, orderId: String, amountCents: Long) = svc.authorize(paymentId, orderId, amountCents)
    fun capture(paymentId: String) = svc.capture(paymentId)
    fun refund(paymentId: String) = svc.refund(paymentId)
}
